src/EventListener/SerializerViewListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\EventListener;
+
+use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Serializes data.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class SerializerViewListener
+{
+    private $serializer;
+
+    public function __construct(SerializerInterface $serializer)
+    {
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * Serializes the data to the requested format.
+     */
+    public function onKernelView(GetResponseForControllerResultEvent $event)
+    {
+        $controllerResult = $event->getControllerResult();
+        $request = $event->getRequest();
+        $format = $request->attributes->get('_api_format');
+
+        if ($controllerResult instanceof Response || !$format) {
+            return;
+        }
+
+        $resourceClass = $request->attributes->get('_resource_class');
+        $collectionOperationName = $request->attributes->get('_collection_operation_name');
+        $itemOperationName = $request->attributes->get('_item_operation_name');
+
```

Comment:
```php
/*
 * @ApiResource(iri="http://schema.org/Person", attributes={
 *     "filters"={"person.search", "person.order", "person.subscribers", "person.customSearch", "person.subscription"},
 *     "normalization_context"={"groups"={"person_default_output", "postal_address_output", "organ_affiliation_default_output", "organ_default_output", "name_default_output", "time_default_output", "person_responsability_default_output", "person_receipt_output", "person_status_default_output", "person_mandate_default_output", "mandate_default_output", "territory_default_output", "bank_default_output"}},
 *     "denormalization_context"={"groups"={"person_default_input", "postal_address_input", "organ_affiliation_default_input", "organ_default_input", "name_default_input", "time_default_input", "person_responsability_default_input", "person_status_default_input", "person_mandate_default_input", "mandate_default_input", "territory_default_input", "bank_default_input"}}
 * }, itemOperations={
 *     "get"={"method"="GET"},
 *     "put"={"method"="PUT"},
 *     "export_subscribers"={
 *        "path"="people_export_subscribers",
 *        "method"="GET",
 *      },
 *     "export_emails"={
 *        "path"="people_export_emails",
 *        "method"="GET",
 *      }
 *     },
 *     collectionOperations={
 *        "get"={"method"="GET"},
 *        "post"={"method"="POST"},
 *        "export_members_email"={
 *          "path"="/people/export_members_email",
 *          "method"="GET",
 *          "pagination_enabled"=false,
 *          "normalization_context"={"groups"={"export_mails_output"}},
 *        }
 *     }
 * )
 */
```

In this case, in my  `export_members_email` i've selected 3 fields. Using this route, with the CsvEncoder in the symfony's PR. I expect to get this three fields, instead of that I get everything